pymedphys/_dicom/anonymise/__init__.py

```diff
@@ -64,7 +65,47 @@ def create_filename_from_dataset(ds, dirpath=""):
     return pjoin(dirpath, "{}.{}.dcm".format(mode_prefix, ds.SOPInstanceUID))
 
 
-def anonymise_dataset(  # pylint: disable = inconsistent-return-statements
+def anonymise_dataset(ds, copy_dataset=True, replacement_strategy=None):
+    r"""A simple tool to anonymise a DICOM dataset.
+
+    You can find the list of DICOM keywords that are included in default
+    anonymisation `here <./identifying_keywords.json>`__.
+    These were drawn from `DICOM Supp 142
+    <https://www.dicomstandard.org/supplements/>`__
+
+    **We do not yet claim conformance to any DICOM Application Level
+    Confidentiality Profile**, but plan to be in a position to do so in the
+    not-to-distant future.
+
+    Parameters
+    ----------
+    ds : ``pydicom.dataset.Dataset``
+        The DICOM dataset to be anonymised.
+
+    copy : ``bool``, optional
+        If ``True``, then a copy of ``ds`` is returned.
+
+    replacement_strategy: ``immutables.Map`` (keys are VR, value is dispatch function), optional
+        If left as the default value of ``None``, the hardcode replacement strategy is used.
+        Also contains delete_private_tags, delete_unknown_tags, and identifying_keywords as keys
+
+    Returns
+    -------
+    ds_anon : ``pydicom.dataset.Dataset``
+        An anonymised version of the input DICOM dataset.
+    """
+
+    if replacement_strategy is None:
+        replacement_strategy = get_copy_of_strategy()
+
+    with replacement_strategy.mutate() as strategy_copy:
+        strategy_copy["copy_dataset"] = copy_dataset
+    replacement_strategy = strategy_copy.finish()
```

Comment:
```suggestion
        replacement_strategy = strategy_copy.finish()
```

...but also, I'd much prefer to simply pass the copy variable through to `_anonymise_dataset` instead of having an inconsistent approach between the two functions.

Reply:
I don't think it's correct to finish and overwrite the original frozenmap within the "with" statement.
For a single change, I could have used a different api that was a one liner, so I'll investigate doing that.